### ISSUE TITLE
"Run Keyword If" enables code injection possiblity

### ISSUE THREAD

**Issue by teemuvesala** [robotframework/robotframework#2856]
Change from test automation to RPA needs new security approach. At RPA world the data might come from he untrusted source (e.g. from web application user). In that case at least the Builtin library should make sure that there isn't any kind of code injection possibility through the variable.  

Below is my PoC robot test. The comparison is taken from the documentation. If you run this with the following command:

> robot -v 'test:print("hellO")' test.robot 

Output is:
```
Test                                                                          
==============================================================================
Test                                                                  hellO
.Hello world
Test                                                                  | PASS |
------------------------------------------------------------------------------
Test                                                                  | PASS |
1 critical test, 1 passed, 0 failed
1 test total, 1 passed, 0 failed
==============================================================================
Output:  /home/tvesala/src/external/robotframework/src/output.xml
Log:     /home/tvesala/src/external/robotframework/src/log.html
Report:  /home/tvesala/src/external/robotframework/src/report.html

```

The output has "hellO" at the screen. The Python code is evaluated. 

The version is: Robot Framework 3.1a2.dev1 (Python 3.5.2 on linux)
(Snapshot from git)

```
*** Keywords ***

Additional Keyword
	Log To Console		Additional keyword

*** Test Cases ***

Test
    Run Keyword If	${test}==0	Additional Keyword
    Log To Console		Hello world
```

**Comment by pekkaklarck** [comment]
This is true but there's really nothing we can do about it. Or do you have ideas?

**Comment by teemuvesala** [comment]
I'll do deeper inspection for this during weekend and think how to change it so that injection possibility is not there. In worst case this mean that eval must be removed and create own parser for conditional statements. It's very difficult to secure eval inputs without white listing the allowed characters. But it would break the functionality really badly. (Remove all ', " and \ characters at least... and even that isn't complete)

**Comment by pekkaklarck** [comment]
First of all, we cannot fully remove the possibility to execute Python code. In addition to `Run Keyword If` and various other keywords handling conditions, there's also `Evaluate` that exists precisely to allow Python code execution.

We could try restricting what code can be executed, but due to Python's dynamic nature that is a *really* big task. There exists, for example, [RestrictedPython](http://restrictedpython.readthedocs.io) that makes it possible, but it is only compatible with CPython and not with other Python implementations we support. I'm also pretty sure there are places where it would cause problems running totally valid and safe code.

The best idea I have is making it possible to use RestrictedPython when using CPython. For backwards compatibility reasons that should be opt-in, enabled with a command line option like `--restrictedeval`.

In general the best way to handle this kind of issues is to not accept inputs from untrusted sources. Even if we were able to restrict Python evaluation, users can change `Remove Directory    ${PATH}    recursive=True` to remove `/` or `c:\`, `Run Process    ${COMMAND}` to run a `rm -rf /`, and so on.

**Comment by pekkaklarck** [comment]
Any new insight about this @teemuvesala?
